src/dune_rules/mdx.ml

```diff
@@ -185,19 +185,32 @@ let syntax =
     [ ((0, 1), `Since (2, 4))
     ; ((0, 2), `Since (3, 0))
     ; ((0, 3), `Since (3, 2))
+    ; ((0, 4), `Since (3, 8))
     ]
 
 let default_files =
   let has_extension ext s = String.equal ext (Filename.extension s) in
   Predicate_lang.Glob.of_pred (has_extension ".md")
 
+let default_files_of_version version =
+  let has_extension ext s = String.equal ext (Filename.extension s) in
+  let is_md = has_extension ".md" in
+  let mld_support_since = (0, 4) in
+  let is_mld =
+    match Syntax.Version.compare version mld_support_since with
+    | Eq | Gt -> has_extension ".mld"
+    | Lt -> fun _ -> false
+  in
+  Predicate_lang.Glob.of_pred (fun f -> is_md f || is_mld f)
```

Comment:
could you use `Predicate_lang.Glob.of_glob` if possible? We're trying to get rid of arbitrary predicates in the glob language.

Reply:
I second this, you could define one predicate per version range and use globs:

```ocaml
let md_files = glob ...
let mld_files = glob ...
match Syntax.Version.compare ... with
| ... -> Predicate_lang.union md_files mld_files
| ... -> md_files
```

Reply:
Done. It's indeed much nicer since the code before was rather inelegant (and was reinventing globs).